dashboard/test/models/sections/section_test.rb

```diff
@@ -71,6 +71,66 @@ class SectionTest < ActiveSupport::TestCase
     end
   end
 
+  test 'changing sharing_disabled to true updates user settings' do
+    student = create :student, sharing_disabled: false
+    section = create :section, sharing_disabled: false
+    assert_creates(Follower) do
+      section.add_student student
+    end
+    section.update(sharing_disabled: true)
+    student.reload
+    assert student.sharing_disabled?
+  end
+
+  test 'changing sharing_disabled to false updates user settings' do
```

Comment:
You may need to reload the `student`. Your instance of student, since it was created on line 86, may have cached attributes.

In other words, try either inserting before line 92 `student.reload` or inlining it:
```ruby
refute student.reload.sharing_disabled?
```

Alternatively you can retrieve the student again from the follower, something like 
```ruby
refute Follower.order(:id).last.student.sharing_disabled?
```
(though I think the first is more concise / readable)